import/index_java/src/org/vufind/index/FormatCalculator.java

```diff
@@ -55,18 +55,23 @@ protected boolean definitelyNotBookBasedOn007(char formatCode) {
      * Determine whether a record cannot be a book due to findings in leader
      * and fixed fields (008).
      *
-     * @param char formatCode
+     * @param char recordType
      * @param ControlField marc008
      * @return boolean
      */
     protected boolean definitelyNotBookBasedOnRecordType(char recordType, ControlField marc008) {
         switch (recordType) {
+            // Computer file
             case 'm':
-                // If this is a computer file containing numeric data, it is not a book:
-                if (getTypeOfComputerFile(marc008) == 'a') {
+                // Check the type of computer file:
+                // If it is 'Document', 'Interactive multimedia', 'Combination',
+                // 'Unknown', 'Other', it could be a book; otherwise, it is not a book:
+                char fileType = get008Value(marc008, 26);
+                if (fileType == 'd' || fileType == 'i' || fileType == 'm' || fileType == 'u' || fileType == 'z') {
+                    return false;
+                } else {
```

Comment:
Since an "else return" is functionally equivalent to a return below the if statement, it might be more concise to remove the else entirely.